src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,213 @@
     @Override
     public void addAll(List<T> list) {
+        for (T listElement: list) {
+            add(listElement);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        if ((index >= size) || (index < 0)) {
+            throw new IndexOutOfBoundsException("Can't get element - index bigger than size");
+        }
+        if (index == 0) {
+            return head.element;
+        }
+        if (index == size - 1) {
+            return tail.element;
+        }
+        return getNode(index).next.element;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        T element;
+        if ((index >= size) || (index < 0)) {
+            throw new IndexOutOfBoundsException("Can't set element - index bigger than size");
+        }
+        if (index == 0) {
+            element = head.element;
+            head.element = value;
+            return element;
+        }
+        if (index == size - 1) {
+            element = tail.element;
+            tail.element = value;
+            return element;
+        }
+        Node<T> node;
+        node = getNode(index).next;
+        element = node.element;
+        node.element = value;
+        return element;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        T element;
+        if ((index >= size) || (index < 0)) {
+            throw new IndexOutOfBoundsException("Can't remove element - index bigger than size");
+        }
+        if ((index == 0) && (size == 1)) {
+            head = null;
+            tail = null;
+            size--;
+            return null;
+        }
+        if (index == 0) {
+            element = head.element;
+            head = head.next;
+            head.prev = null;
+            size--;
+            return element;
+        }
+        if (index == size - 1) {
+            element = tail.element;
+            tail = tail.prev;
+            tail.next = null;
+            size--;
+            return element;
+        }
+        Node<T> oldNode;
+        oldNode = getNode(index).next;
+        element = oldNode.element;
+        unLink(oldNode);
+        size--;
+        return element;
     }
 
     @Override
     public boolean remove(T object) {
+        Node<T> node = head;
+        boolean equal = (head.element == null) ? (object == null) : (head.element.equals(object));
+        if ((size == 1) && equal) {
+            head = null;
+            tail = null;
+            size--;
+            return true;
+        }
+        if (equal) {
+            head.next.prev = null;
+            head = head.next;
+            size--;
+            return true;
+        }
+        for (int i = 1; i < size; i++) {
+            node = node.next;
+            equal = (node.element == null) ? (object == null) : (node.element.equals(object));
+            if (equal) {
+                unLink(node);
+                size--;
+                return true;
+            }
```

Comment:
we need to find node by value 
```suggestion
        Node<T> node = head;
        while (node != null) {
            if (node.element == object || node.element != null && node.element.equals(object)) {
                unlink(node);
                return true;
            }
            node = node.next;
        }
        return false;

```